### PR TITLE
AP-455: Add generic `get_dataset` method

### PR DESCRIPTION
**Pull request by rlmv**
Add a `get_dataset` method that tries to get a dataset by name or by ID.

To enable https://github.com/Blackfynn/agent/pull/89

### CODE REVIEW

File: src/bf/api/client/mod.rs [hunk 1]
```diff
@@ -488,6 +488,26 @@ impl Blackfynn {
         }))
     }
 
+    /// Get a dataset by ID or by name.
+    pub fn get_dataset<N: Into<String>>(&self, id_or_name: N) -> bf::Future<response::Dataset> {
```
Comment (mattusifer) [inline src/bf/api/client/mod.rs]: can we just add a return type here? I think it helps make the signature more readable.

Comment (rlmv) [reply src/bf/api/client/mod.rs]: What do you mean by that? Something in addition to `bf::Future<response::Dataset>`?

**Comment by rlmv** [comment]
@mikeswoods I'm going to let #32 take care of the version bump - the agent codebase is not currently compiling with `blackfynn-rust` `master` right now because of changes to the upload models so we need to wait until those changes hit the agent anyway.

File: src/bf/api/client/mod.rs (hunk 1 above)
Comment (mattusifer) [reply src/bf/api/client/mod.rs]: weird.. I dont know why I didn't see that there when I initially commented on this

**Pull request merged**